lib/castle/client.rb

```diff
@@ -2,14 +2,13 @@
 
 module Castle
   class Client
-    attr_accessor :do_not_track, :api
+    attr_accessor :api
 
     def initialize(request, response)
       @do_not_track = false
-      cookie_id = extract_cookie(request, response)['__cid'] || ''
-      ip = request.ip
-      headers = header_string(request)
-
+      cookie_id = Extractors::Cookie.new(request).extract(response, '__cid')
```

Comment:
FYI, this will soon be replaced by `client_id`, with logic like this:

```
client_id =
  request.cookies['__cid'] ||
  request.headers['X-Castle-Client-Id']
```

So maybe call it `Extractors::ClientId`?